src/handler.ts

```diff
@@ -139,6 +179,11 @@ abstract class AbstractFileHandler implements IMessageFilter {
   protected abstract populateWidget(widget: Widget, model: IContentsModel): Promise<void>;
 
   /**
+   * Get the contents moedl for a widget.
+   */
+  protected abstract getWidgetContents(widget: Widget): Promise<IContentsModel>;
```

Comment:
I think it would be better if the `getWidgetContents` and `populateWidget` had names that were inverses of each other, like `getState` and `setState` or something (not sure what the right name should be).